theano/sandbox/cuda/opt_util.py

```diff
@@ -100,9 +101,7 @@ def opt(node):
                     # other cases are too complex for now
                     return None
                 if W.broadcastable != targ.inputs[out_in].broadcastable:
-                    # Would need to explicitly tile the output to fill
-                    # the full shape here.  Disable for now.
-                    return None
+                    W = broadcast_like(W, targ.inputs[out_in], node.fgraph)
```

Comment:
I'm not sure this give a speed up and it make the graph more complicated. What make you think adding an alloc in the graph will be faster then having an elemwise add? Did you time this?